src/backend/metal/src/command.rs

```diff
@@ -1963,14 +1963,78 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
         T: IntoIterator,
         T::Item: Borrow<com::BufferCopy>,
     {
-        let commands = regions.into_iter().map(|region| {
-            soft::BlitCommand::CopyBuffer {
-                src: src.raw.clone(),
-                dst: dst.raw.clone(),
-                region: region.borrow().clone(),
+        let inner = unsafe { &mut *self.inner.get() };
+        let compute_pipe = self.shared.service_pipes
+            .lock()
+            .unwrap()
+            .get_copy_buffer()
+            .to_owned();
+
+        let mut blit_commands = Vec::new();
+        let mut compute_commands = vec![
+            soft::ComputeCommand::BindPipeline(compute_pipe),
+        ];
+        let wg_size = MTLSize {
+            width: 64,
+            height: 1,
+            depth: 1,
+        };
+
+        for region in  regions {
+            let r = region.borrow();
+            if r.size % 4 == 0 {
+                blit_commands.push(soft::BlitCommand::CopyBuffer {
+                    src: src.raw.clone(),
+                    dst: dst.raw.clone(),
+                    region: r.clone(),
+                });
+            } else {
+                // not natively supported, going through compute shader
+                assert_eq!(0, r.size >> 32);
+                let word_count = (r.size + 3) / 4;
+                let copy_data = self.shared.device.new_buffer_with_data(
+                    &r.size as *const u64 as *const _,
```

Comment:
Do we need to do `&(r.size as u32) as *const u32` here or can we assume this is safe?

Reply:
I assume this is safe

Reply:
Why don't we get a void pointer to a `u64`?

Reply:
let's do it, yeah

Reply:
The buffer is supposed to be `u32` isn't it? I thought it would be:
```rust
&(r.size as u32) as *const u32 as *const _,
mem::size_of::<u32>() as _
```